src/moscot/problems/_utils.py

```diff
@@ -48,34 +49,42 @@ def handle_joint_attr(
 
 
 def handle_cost(
-    xy: Optional[Mapping[str, Any]] = None,
-    x: Optional[Mapping[str, Any]] = None,
-    y: Optional[Mapping[str, Any]] = None,
+    xy: Mapping[str, Any] = types.MappingProxyType({}),
+    x: Mapping[str, Any] = types.MappingProxyType({}),
+    y: Mapping[str, Any] = types.MappingProxyType({}),
     cost: Optional[Union[CostFn_t, Mapping[str, CostFn_t]]] = None,
+    cost_kwargs: Union[Mapping[str, Any], Mapping[str, Mapping[str, Any]]] = types.MappingProxyType({}),
     **_: Any,
-) -> Tuple[Optional[Mapping[str, Any]], Optional[Mapping[str, Any]], Optional[Mapping[str, Any]]]:
+) -> Tuple[Dict[str, Any], Dict[str, Any], Dict[str, Any]]:
+    x = dict(x)
+    y = dict(y)
+    xy = dict(xy)
     if cost is None:
         return xy, x, y
-    if isinstance(cost, str):
-        if xy is not None and "cost" not in xy:
-            xy = dict(xy)
-            xy["cost"] = cost
-        if x is not None and "cost" not in x:
-            x = dict(x)
+    if isinstance(cost, str):  # if cost is a str, we use it in all terms
+        if xy and ("x_cost" not in xy or "y_cost" not in xy):
+            xy["x_cost"] = xy["y_cost"] = cost
+        if x and "cost" not in x:
             x["cost"] = cost
-        if y is not None and "cost" not in y:
-            y = dict(y)
+        if y and "cost" not in y:
             y["cost"] = cost
-        return xy, x, y
-    if isinstance(cost, Mapping):
-        if xy is not None and "cost" not in xy:
-            xy = dict(xy)
-            xy["cost"] = cost["xy"]
-        if x is not None and "cost" not in x:
-            x = dict(x)
+    elif isinstance(cost, Mapping):  # if cost is a dict, the cost is specified for each term
+        if xy and ("x_cost" not in xy or "y_cost" not in xy):
+            xy["x_cost"] = xy["y_cost"] = cost["xy"]
+        if x and "cost" not in x:
             x["cost"] = cost["x"]
-        if y is not None and "cost" not in y:
-            y = dict(y)
+        if y and "cost" not in y:
             y["cost"] = cost["y"]
-        return xy, x, y
-    raise TypeError(type(cost))
+    else:
+        raise TypeError(type(cost))
+    if xy and cost_kwargs:  # distribute the cost_kwargs, possibly explicit to x/y/xy-term
+        items = (
+            cost_kwargs["xy"].items() if "xy" in cost_kwargs else cost_kwargs.items()
+        )  # extract cost_kwargs explicit to xy-term if possible
```

Comment:
I think if you put this comment above `items` definition, `black` will format it nicely (currently, the line seems to be long).